File: rust/src/models/config_value.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{
    boolean_config_value::BooleanConfigValue, json_config_value::JsonConfigValue,
    number_config_value::NumberConfigValue, text_config_value::TextConfigValue,
};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
#[serde(tag = "kind")]
pub enum ConfigValue {
    #[serde(rename = "NUMBER")]
    Number(NumberConfigValue),
    #[serde(rename = "BOOLEAN")]
    Boolean(BooleanConfigValue),
    #[serde(rename = "TEXT")]
    Text(TextConfigValue),
    #[serde(rename = "JSON")]
    Json(JsonConfigValue),
}

impl Default for ConfigValue {
    fn default() -> Self {
        Self::Number(NumberConfigValue::default())
    }
}

File: python/meteroid/models/config_value.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel, TaggedUnionModel
from .boolean_config_value import BooleanConfigValue
from .json_config_value import JsonConfigValue
from .number_config_value import NumberConfigValue
from .text_config_value import TextConfigValue


@dataclasses.dataclass
class ConfigValue(TaggedUnionModel):
    """A static, typed configuration value carried by a Config entitlement. Resolved synchronously
    through the entitlement hierarchy — no metric, no usage counter."""

    _DISCRIMINATOR: t.ClassVar[str] = "kind"
    _DISCRIMINATOR_ATTR: t.ClassVar[str] = "kind"
    _CONTENT_ATTR: t.ClassVar[str] = "content"
    _CONTENT_KEY: t.ClassVar[t.Optional[str]] = None
    _VARIANTS: t.ClassVar[t.Mapping[str, t.Optional[t.Type[BaseModel]]]] = {
        "NUMBER": NumberConfigValue,
        "BOOLEAN": BooleanConfigValue,
        "TEXT": TextConfigValue,
        "JSON": JsonConfigValue,
    }

    kind: t.Literal[
        "NUMBER",
        "BOOLEAN",
        "TEXT",
        "JSON",
    ]
    content: t.Union[
        NumberConfigValue,
        BooleanConfigValue,
        TextConfigValue,
        JsonConfigValue,
    ]
